Dockerfile

```diff
@@ -0,0 +1,11 @@
+FROM --platform=$BUILDPLATFORM gradle:jdk17 as builder
+WORKDIR /build
+COPY build.gradle settings.gradle gradle.properties ./
+COPY .git .git
+COPY src src
+RUN gradle --no-daemon build
+
+FROM --platform=$TARGETPLATFORM eclipse-temurin:21-jre-alpine
+WORKDIR /app
+COPY --from=builder /build/build/libs/ViaProxy-*.jar ViaProxy.jar
+ENTRYPOINT ["java", "-jar", "ViaProxy.jar"]
```

Comment:
These java args are pretty barebones. You'd also want controllable RAM and gc flags.

Reply:
The default JVM arguments here would be G1GC with 2 GB of RAM allocated to it as far as i'm aware, however if needed then optimizations should be thought upon.

Reply:
I think i should be able to make the args passed to java controllable with for example a JAVA_OPTS env variable